test/e2e/basic/metrics_cert_rotation.go

```diff
@@ -0,0 +1,77 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	ntoconfig "github.com/openshift/cluster-node-tuning-operator/pkg/config"
+	util "github.com/openshift/cluster-node-tuning-operator/test/e2e/util"
+)
+
+var _ = ginkgo.Describe("[basic][metrics] Node Tuning Operator certificate rotation", func() {
+
+	ginkgo.Context("TLS certificate rotation", func() {
+		ginkgo.It("delete certificate Secret and check that the server restarts with latest certificate", func() {
+			const (
+				pollInterval = 5 * time.Second
+				waitDuration = 5 * time.Minute
+			)
+
+			// Delete the existing secret to force certificate rotation
+			_, _, err := util.ExecAndLogCommand("oc", "delete", "-n", ntoconfig.OperatorNamespace(), "secret/node-tuning-operator-tls")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			ginkgo.By("sleeping 5s to wait for Secret to be refreshed")
+			time.Sleep(5 * time.Second)
```

Comment:
Can we be reasonably sure the secrets are refreshed within in 5 seconds?  Is it possible to do this in a cleaner way without the sleep?  Ignore if it is too hard to implement.

Reply:
I rewrote this part into a function that will delete it, then check every 2 seconds for up to 2 minutes whether it is recreated. It seems to usually be back by the second check. I think this is "better" so we don't have to rely on it being refreshed within 5 seconds, but it will also not wait as long most of the time.

(Will update PR once I address the error handling in the server restart)

Reply:
Pushed the change to this part.